### DISCUSSION TITLE
Check for whether a search param is explicitly *not* set

### DISCUSSION THREAD

**Discussion by MayaRainer** [TanStack/router#2566] [Q&A] [answered]
I'm writing a component that renders a list of tab links, only one of which should be active at a time. In a specific use case, one of these links should only be active when a search parameter set by the others _isn't_ set.

In #1969, the `Link` behaviour was changed to treat an explicitly `undefined` property the same as the absence of the property - but that seems to leave us without a way of explicitly validating their absence. Using `exact` _kind of_ works, but only when no other search parameters are involved.

Example (these are actually rendered in a list as it's a reusable component, but will just add two each here for simplicity):
```tsx
<Link search={{ param: "value" }}>...
<Link search={{ param: undefined }}>... // will be active even when `param` is set
```

```tsx
<Link search={{ param: "value" }} activeOptions={{ exact: true }}>...
<Link search={{ param: undefined }} activeOptions={{ exact: true }}>... // will not be active even when `param` is set, but also not be active when another search param like `page` is set
```

```tsx
const location = useLocation();
...
<Link
  search={{ ...location.search, param: "value" }}
  activeOptions={{ exact: true }}
>...
<Link
  search={{ ...location.search, param: undefined }}
  activeOptions={{ exact: true }}
>... // will not be active even when `param` is set, and ignores other search params, but makes it impossible to remove other search params like `page`
```

Lmk if I'm missing something, but I haven't been able to find something that really works for this use case. Thanks in advance!

**Comment by schiller-manuel** [comment]
we could make the meaning of `undefined` in`activeOptions` configurable

**Reply by schiller-manuel** [reply, marked as answer]
I implemented this in https://github.com/TanStack/router/pull/2592

usage:

```tsx
<Link search={{ param: undefined }} activeOptions={{ explicitUndefined: true }}>
```
